apps/backoffice-v2/src/domains/business-reports/hooks/mutations/useCreateBusinessReportBatchMutation/useCreateBusinessReportBatchMutation.tsx

```diff
@@ -0,0 +1,45 @@
+import { t } from 'i18next';
+import { toast } from 'sonner';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { HttpError } from '@/common/errors/http-error';
+import { TBusinessReportType } from '@/domains/business-reports/types';
+import { createBatchBusinessReport } from '@/domains/business-reports/fetchers';
+import { useCustomerQuery } from '@/domains/customer/hook/queries/useCustomerQuery/useCustomerQuery';
+
+export const useCreateBusinessReportBatchMutation = ({
+  reportType,
+  onSuccess,
+}: {
+  reportType: TBusinessReportType;
+  onSuccess?: <TData>(data: TData) => void;
+}) => {
+  const queryClient = useQueryClient();
+
+  const { data: customer } = useCustomerQuery();
+
+  return useMutation({
+    mutationFn: (merchantSheet: string) =>
+      createBatchBusinessReport({
+        reportType,
+        merchantSheet,
+        isExample: customer?.config?.isExample ?? false,
+      }),
+    onSuccess: data => {
+      void queryClient.invalidateQueries();
+
+      toast.success(t(`toast:batch_business_report_creation.success`));
+
+      onSuccess?.(data);
+    },
+    onError: (error: Error) => {
+      if (error instanceof HttpError && error.code === 400) {
+        toast.error(error.message);
+
+        return;
+      }
```

Comment:
Do we have this in other places? I remember we handle this globally.

Reply:
The only other place we have this (or anything similar, at least not that I can find) is the `useCreateBusinessReportMutation` (which was the base for my code).